### ISSUE TITLE
Stop historical processing on indexer redeployment

### ISSUE THREAD

**Issue by Ishatt** [near/queryapi#200]
Description

When saving the new code, user has an option to “start over” – wipe the database, replace the code, and restart historical processing from the selected block

Coordinator should detect this option in registry indexing

Coordinator should purge the SQS queue using the group id of this indexer, or the runner should ignore all messages. This might need to be refactored to use Redis Streams instead (@Eduardo Ohe will investigate how Runners can be hooked to Redis Streams https://redis.io/docs/data-types/streams-tutorial/#streams-observability )

Coordinator should send a re-provision message to the runner so that it deletes the database schema for this indexer and provisions it again. Runner should update the state of the indexer and keep user updated in the logs

Coordinator should re-start the historical thread from the provided block height

**Comment by morgsmccauley** [comment]
Historical processing is offloaded to a `tokio` async task, before starting the next historical process we must kill the existing. Initially, I thought it would be possible to cancel the this task from the outside with either:
1. something similar to`std::thread::abort`, or by
2. `drop`ing the `JoinHandle`

But neither of these are feasible. 1. is not supported, and 2. will only abort the task if the underlying process yields control at various points, which it seems like it does not.

Instead, I'm going to have to refactor the `histroical_block_processing()` function so that it intermittently checks  some form of `stop_token` which can be set from the outside. We can use [tokio::select](https://tokio.rs/tokio/tutorial/select) to interleave the actual historical processing logic, and the stop token check.

**Comment by morgsmccauley** [comment]
Another issue, which can be resolved as part of this ticket, is that historical processing messages are written only _after_ we have processed all relevant blocks. This causes a large delay between 'starting' a historical process, and the point where the blocks start executing.

We should refactor the historical process so that it publishes Redis messages _while_ it is processing them, so that execution can begin almost immediately.
